controllers/bootstrap/init.go

```diff
@@ -111,6 +111,21 @@ func (b *Bootstrap) InitResources(manualManagement bool) error {
 		return err
 	}
 
+	// Check Saas Deployment
+	controlNs := b.MasterNamespace
+	saasEnable, err := checkSaas(b.Reader)
+	if err != nil {
+		return err
+	}
+	if saasEnable {
+		klog.Info("Saas Deployment Enabled for Common Services")
+		controlNs = util.GetControlNs(b.Reader)
```

Comment:
We need a check to make sure if controlNs is empty

Reply:
Did you mean that even if the SaaS deployment is enabled, the `control namespace` in `common-service-maps` ConfigMap may be empty, then we should have a default value like what we did for `MasterNamespace` which is `ibm-common-services`?

Reply:
Yes, exactly.

Reply:
ah i see, Updated. Please see the comment here
https://github.com/IBM/ibm-common-service-operator/pull/446#discussion_r602811791

Reply:
Can users not install normally (shared CS) on SaaS? So SaaS installation automatically means multiple CS instances? If they can still install normally (shared CS), then control namespace should be the same as master namespace.

Reply:
@bitscuit 
As we discussed today, end-user or SaaS automatically can use `multiInstanceEnabled` to decide install common service into shared or multi-instance mode.

Also, we can't set the control namespace as one of the master namespaces. Since each tenant scope needs to have permission to control namespace. If the control namespace is within one scope, that will cause there is a permission overlap between tenant scope.  We can set a check for it.

Reply:
for the SaaS deployment, we should have a separate control namespace which distinguishes from any master namespace for the cluster scope resource.

It is because the one item in `namespaceMapping` list has its own scope like `cp4i`, `cp4mcm` and `cp-common-service`, we should not include control namespace in any of them.

```
controlNamespace: ibm-cs-control
    namespaceMapping:
     - requested-from-namespace:
       - cp4i
       - cp4mcm
       map-to-common-service-namespace: cp-common-service
     - requested-from-namespace:
       - iaf
       - watson
       map-to-common-service-namespace: watson-common-service
```